### PR TITLE
Jbau/lms preview

### PR DESCRIPTION
**Pull request by jbau**

### CODE REVIEW

File: playbooks/secure_example/vars/edxapp_prod_vars.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ---
-lms_auth_config:
+#Use YAML references (& and *) and hash merge <<: to factor out shared settings
+#see http://atechie.net/2009/07/merging-hashes-in-yaml-conf-files/
+lms_auth_config:  &lms_auth
```
Comment (jbau) [inline playbooks/secure_example/vars/edxapp_prod_vars.yml:4]: here be magical YAML syntax

File: playbooks/secure_example/vars/edxapp_prod_vars.yml [hunk 2]
```diff
@@ -144,4 +146,23 @@ cms_env_config:
     'DISABLE_COURSE_CREATION': false
     'SEGMENT_IO': false
 
+lms_preview_auth_config:
```
Comment (jbau) [inline playbooks/secure_example/vars/edxapp_prod_vars.yml:149]: this is what the lms_preview is reduced to

File: playbooks/roles/lms-preview/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#  - nginx/tasks/main.yml
+---
+- name: create lms application config
+  template: src=env.json.j2 dest=$app_base_dir/lms-preview.env.json
+  tags:
+  - lms-preview
+  - lms-preview-env
+
+- name: create lms auth file
+  template: src=auth.json.j2 dest=$app_base_dir/lms-preview.auth.json
```
Comment (jarv) [inline playbooks/roles/lms-preview/tasks/main.yml:13]: is it valid to use jinja syntax here for consistency `{{ app_base_dir }}/lms-preview.auth.json`?

File: playbooks/secure_example/vars/edxapp_prod_vars.yml (hunk 1 above)
Comment (jarv) [inline playbooks/secure_example/vars/edxapp_prod_vars.yml:4]: nice, I was thinking about doing something similar and have heard of other ansible users doing similar ~~awful~~ awesome things

**Comment by jarv** [comment]
good to merge

**Comment by jbau** [comment]
thx!  We'll move over to jinja synax as we go.

**Pull request merged**